corehq/apps/sso/tasks.py

```diff
@@ -99,3 +109,53 @@ def send_idp_cert_expires_reminder_emails(num_days):
                 f"Failed to send cert reminder email for IdP {idp}: {exc!s}",
                 exc_info=True,
             )
+
+
+@periodic_task(run_every=crontab(minute=0, hour=2), acks_late=True)
+def auto_deactivate_removed_sso_users():
+    for idp in IdentityProvider.objects.filter(
+        enable_user_deactivation=True,
+        idp_type=IdentityProviderType.AZURE_AD
+    ).all():
+        # Fetch a list of users usernames that are members of the idp
+        idp_users = idp.get_all_members_of_the_idp()
+        # Fetch a list of active WebUser usernames that are members of project spaces associated with the idp
+        domains = idp.owner.get_domains()
+        web_user_in_account = set()
+        for domain in domains:
+            [web_user_in_account.add(user.username) for user in WebUser.by_domain(domain)]
+
+        # Get a list of SSO Exempt users
+        authenticated_domains = AuthenticatedEmailDomain.objects.filter(identity_provider=idp)
+        sso_exempt_users = UserExemptFromSingleSignOn.objects.filter(email_domain__in=authenticated_domains)
+        exempt_usernames = sso_exempt_users.values_list('username', flat=True)
+
+        if len(idp_users) == 0 and len(web_user_in_account) - len(exempt_usernames) > 3:
+            # Send email
+            subject = _("Temporarily skipped automatic deactivation of Web Users")
+            recipient = idp.owner.enterprise_admin_emails
+            body = _("we have temporarily skipped automatic deactivation of Web Users because we are receiving "
```

Comment:
Similar to the previous suggestion for updating the `body` text in `get_all_members_of_the_idp`

```
Dear enterprise administrator,

We have temporarily skipped automatic deactivate of SSO Web Users because we are receiving an empty list of users from your Microsoft Entra ID instance.

Please check the configuration of the Remote User Management settings in your Identity Provider.

If you have any questions, or if this issue persists, please don’t hesitate to contact {{ contact_email }}.
Thank you for your use and support of CommCare.

Best regards,

The CommCare HQ Team
```

Reply:
this email bit can also be its own utility abstracted away

Reply:
Thank you Biyeun. Improved in 9128522fe0edb09107879367efc6cfb52478079b